rules/otp.bzl

```diff
@@ -348,3 +350,108 @@ STD_OTP_OVERLAYS = STD_OTP_OVERLAYS_WITHOUT_SECRET_PARTITIONS + [
     "//hw/ip/otp_ctrl/data:otp_json_secret1",
     "//hw/ip/otp_ctrl/data:otp_json_secret2_unlocked",
 ]
+
+def otp_hex(v):
+    return hex(v)
+
+def otp_bytestring(byte_list):
+    bytestring = "".join(reversed([hex_digits(byte, width = 8) for byte in byte_list]))
+    return "0x{}".format(bytestring)
+
+def _parse_str_list(s):
+    """Parse a string with comma delimiters as a list of strings, ignoring spaces.
+
+    This function is a workaround when lists need to be provided as arguments in
+    a compact form in a BUILD file. Leaving the argument as a list causes
+    buildifier to reformat it onto multiple lines whereas a string is kept as
+    one line. This function can be removed once buildifier supports exempting
+    regions from auto-formatting.
```

Comment:
ah, that's the reason then. IIRC you mentioned that you can exclude files from linting but not formatting -- I think `buildifier` got it backwards.
I'd suggest we should just live with the bad format but good data as opposed to bad data but good format.

Reply:
@cfrantz and I had a discussion about this [above](https://github.com/lowRISC/opentitan/pull/18439#discussion_r1190333924), and the gist of this is that for the alert class configuration, this would be significantly harder to read and more error-prone when we have more alerts. Given that the `otp_alert_classification` has strict checks on parsed string, we think that this tradeoff is fine.

Consider two versions with just 7 entries:

String version:
```
"OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(
    alert_list = EARLGREY_ALERTS,
    # The ordering is             prod, prod_end, dev, rma
    default = "                      X,        X,   X,   X",
    gpio_fatal_fault = "             X,        X,   X,   X",
    i2c0_fatal_fault = "             X,        X,   X,   X",
    otp_ctrl_fatal_macro_error = "   X,        X,   X,   X",
    pattgen_fatal_fault = "          X,        X,   X,   X",
    rv_timer_fatal_fault = "         X,        X,   X,   X",
    spi_device_fatal_fault = "       X,        X,   X,   X",
    uart0_fatal_fault = "            X,        X,   X,   X",
),
```

List version:
```
"OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(
    alert_list = EARLGREY_ALERTS,
    default = [
        "X",
        "X",
        "X",
        "X",
    ],
    gpio_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
    i2c0_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
    otp_ctrl_fatal_macro_error = [
        "X",
        "X",
        "X",
        "X",
    ],
    pattgen_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
    rv_timer_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
    spi_device_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
    uart0_fatal_fault = [
        "X",
        "X",
        "X",
        "X",
    ],
)
```

Reply:
I see, thanks. Not for this PR but I wonder how a dict would look (along with a helper function that doesn't rely on tokenization). sthg like {0: "foo", 1: "bar"} but buildifier would probably _format_ that too.